### PR TITLE
Server: Support for request compression.

### PR DESCRIPTION
**Pull request by gianm**

### CODE REVIEW

**Comment by fjy** [comment]
👍

File: server/src/main/scala/com/metamx/tranquility/server/http/TranquilityServlet.scala [hunk 1]
```diff
@@ -154,6 +157,17 @@ class TranquilityServlet(
     }
   }
 
+  private def getRequestDecompressor(): InputStream => InputStream = {
+    request.header("Content-Encoding") match {
+      case Some("gzip") | Some("x-gzip") =>
+        in => new GZIPInputStream(in)
```
Comment (drcrallen) [inline server/src/main/scala/com/metamx/tranquility/server/http/TranquilityServlet.scala]: FYI: http://bugs.java.com/bugdatabase/view_bug.do?bug_id=7036144

com.metamx.common.CompressionUtils#gzipInputStream has a workaround for it

Comment (gianm) [inline server/src/main/scala/com/metamx/tranquility/server/http/TranquilityServlet.scala]: thanks @drcrallen, updated this to use that function.

**Comment by gianm** [comment]
closing to free up travis resources until #176 is sorted out.

**Pull request merged**